src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -589,7 +596,8 @@ public static void sleep(Duration duration) throws InterruptedException {
             if (currentThread() instanceof VirtualThread vthread) {
                 vthread.sleepNanos(nanos);
             } else {
-                sleep0(nanos);
+                long millis = NANOSECONDS.toMillis(nanos);
+                sleepImpl(millis, (int)(nanos - MILLISECONDS.toNanos(millis)));
```

Comment:
The conversion functions `toMillis()` and `toNanos()` clamp large values (although that shouldn't happen given `convert()` would have alreay done so). We don't need the expense of calling `toMillis()` and `toNanos()`; suggest:
```java
                sleepImpl(nanos / 1_000_000L, (int)(nanos % 1_000_000L));
```

Reply:
Updated